### PR TITLE
harvesting tests in Github Actions

### PR DESCRIPTION
**Pull request by eriktks**
Enabled unit testing for harvesting part on Github Actions cf issue #490 . I do not know how to test if it works (other than merging).

### CODE REVIEW

**Comment by eriktks** [comment]
The initial tests (Actions/Harvesting tests) fail but that is because pytest cannot find any test files. The file name problem of the test files that causes this (*test.py -> *_test.py) was fixed in this PR.

File: .github/workflows/harvesting_tests.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    - name: Install dependencies
+      run: |
+        cd frontend
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+        pip install pytest==5.* py-w3c
+    - name: Tests with pytest (unit tests only)
+      run: |
+        cd harvesting
+        PYTHONPATH=. pytest -k "unit_test"
```
Comment (jspaaks) [inline .github/workflows/harvesting_tests.yml:33]: On my system, this works:

```suggestion
        pytest *_unittest.py 
```

File: .github/workflows/harvesting_tests.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        cd frontend
```
Comment (jspaaks) [inline .github/workflows/harvesting_tests.yml:26]: expected to see `harvesting` here

File: .github/workflows/harvesting_tests.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+        pip install pytest==5.* py-w3c
```
Comment (jspaaks) [inline .github/workflows/harvesting_tests.yml:29]: `harvesting` has a requirements-dev.txt that we should use; probably dep versions in there needs updating

File: harvesting/util.py [hunk 4]
```diff
@@ -69,9 +69,11 @@ def db_connect():
                                )[os.environ.get('DATABASE_NAME')]
 
 def rate_limit_reached(requests_response):
+    if requests_response is None:
+        return True
     rate_limit_remaining = requests_response.headers.get('x-ratelimit-remaining', None)
     if rate_limit_remaining is None:
-        return False
+        return True
```
Comment (jspaaks) [inline harvesting/util.py:76]: I think this should be False

**Review by jspaaks — CHANGES_REQUESTED** [review]

File: .github/workflows/harvesting_tests.yml (hunk 1 above)
Comment (eriktks) [reply .github/workflows/harvesting_tests.yml:33]: test files had incorrect names; resolved by committing name changes

File: .github/workflows/harvesting_tests.yml (hunk 3 above)
Comment (eriktks) [reply .github/workflows/harvesting_tests.yml]: Done

File: .github/workflows/harvesting_tests.yml (hunk 2 above)
Comment (eriktks) [reply .github/workflows/harvesting_tests.yml]: Changed

File: harvesting/util.py (hunk 4 above)
Comment (eriktks) [reply harvesting/util.py]: Changed

**Review by jspaaks — APPROVED** [review]
:trophy:

**Comment by eriktks** [comment]
Thanks for the review and the award!

**Pull request merged**